src/basic/user-util.c

```diff
@@ -460,13 +461,13 @@ int get_shell(char **_s) {
 
 int reset_uid_gid(void) {
 
-        if (setgroups(0, NULL) < 0)
+        if (setgroups(0, NULL) < 0 && (errno != EPERM || (detect_container() <= 0)))
                 return -errno;
 
-        if (setresgid(0, 0, 0) < 0)
+        if (setresgid(0, 0, 0) < 0 && (errno != EPERM || (detect_container() <= 0)))
                 return -errno;
 
-        if (setresuid(0, 0, 0) < 0)
+        if (setresuid(0, 0, 0) < 0 && (errno != EPERM || (detect_container() <= 0)))
                 return -errno;
```

Comment:
not sure i grok this, but what precisely is failing here in your case? why shouldn't we be able to reset our creds to root?